### PR TITLE
Issue 17052 json serializer improvements for page resource

### PR DESCRIPTION
**Pull request by nollymar**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05180UpdateFriendlyNameField.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.startup.StartupTask;
+import com.dotmarketing.util.Logger;
+
+/**
+ * @author nollymar
+ */
+public class Task05180UpdateFriendlyNameField implements StartupTask {
+
+    private final String UPDATE_FIELD = "UPDATE field set velocity_var_name='" + FRIENDLY_NAME_FIELD
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05180UpdateFriendlyNameField.java]: I think this should be more specific, joining to the structure table and adding a `where structuretype=5`

**Review by wezell — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05180UpdateFriendlyNameField.java (hunk 1 above)
Comment (wezell) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05180UpdateFriendlyNameField.java]: meaning, only update the `friendlyname` fields on htmlpages

Comment (nollymar) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05180UpdateFriendlyNameField.java]: good catch!

Comment (nollymar) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05180UpdateFriendlyNameField.java]: done

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java [hunk 2]
```diff
@@ -624,10 +626,10 @@ public void setLongProperty(com.dotcms.contenttype.model.field.Field field,long
 	public long getLongProperty(String fieldVarName) throws DotRuntimeException {
 		try{
 	    final Object test = map.get(fieldVarName);
-	    return test == null 
-	        ? 0 
-	            : test instanceof String 
-	            ? Long.parseLong((String) test)  
+	    return test == null
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java]: I think this could be change for ConversionUtils.toLong(test, 0)

**Review by dsilvam — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**